torch/csrc/jit/passes/utils/subgraph_utils.cpp

```diff
@@ -394,6 +394,31 @@ Node* createSingletonSubgraphAndUpdateAliasing(
       });
 }
 
+std::string truncateStrWithHash(const std::string& s, size_t maxlen) {
+  if (s.size() <= maxlen) {
+    return s;
+  }
+  std::stringstream truncated;
+  truncated << s.substr(0, maxlen);
+  truncated << "_" << std::hash<std::string>{}(s) << "_";
+  return truncated.str();
+}
+
+std::string generateNameForGraph(
+    std::shared_ptr<Graph> graph,
```

Comment:
Oh also the lint about shared_ptr<Graph> -> const Graph& is probably a good idea.